core/src/main/scala/cats/data/Kleisli.scala

```diff
@@ -180,20 +180,20 @@ private[data] sealed abstract class KleisliInstances3 extends KleisliInstances4
   }
 }
 
-private[data] sealed abstract class KleisliInstances4 {
+private[data] sealed abstract class KleisliInstances4 extends KleisliInstances5 {
 
-  implicit def catsDataMonadReaderForKleisli[F[_]: Monad, A]: MonadReader[Kleisli[F, A, ?], A] =
-    new MonadReader[Kleisli[F, A, ?], A] {
-      def pure[B](x: B): Kleisli[F, A, B] =
-        Kleisli.pure[F, A, B](x)
-
-      def flatMap[B, C](fa: Kleisli[F, A, B])(f: B => Kleisli[F, A, C]): Kleisli[F, A, C] =
-        fa.flatMap(f)
+  implicit def catsDataMonadReaderForKleisli[F[_], A](implicit ev: Monad[F]): MonadReader[Kleisli[F, A, ?], A] =
+    new KleisliMonadReader[F, A] {
+      def F0: Monad[F] = ev
+    }
+}
 
-      val ask: Kleisli[F, A, A] = Kleisli(Monad[F].pure)
+private[data] sealed abstract class KleisliInstances5 {
 
-      def local[B](f: A => A)(fa: Kleisli[F, A, B]): Kleisli[F, A, B] =
-        Kleisli(f.andThen(fa.run))
+  implicit def catsDataMonadRecForKleisli[F[_], A](implicit ev0: Monad[F], ev1: MonadRec[F]): MonadRec[Kleisli[F, A, ?]] =
```

Comment:
Should only need `MonadRec[F]` since that implies `Monad[F]`. And for single implicit parameters I believe the Cats style is:

```scala
catsDataMonadRecForKleisli[F[_], A](implicit F: Monad[F]): MonadRec[Kleisli[F, A, ?]] = ...
```